### DISCUSSION TITLE
How to properly use React's cache function

### DISCUSSION THREAD

**Discussion by m1yon** [vercel/next.js#49767] [App Router]
In order for caching/deduping to work, do you have to call the same function returned from cache each time? Or is sending cache the same function good enough?

Example:
```tsx
const preload = ({ params }: { params: { shortId: string; slug: string } }) => {
  cache(getPersonById)({
    shortId: params.shortId,
    slug: params.slug,
  })
}

// different component
cache(getPersonById)({
  shortId: params.shortId,
  slug: params.slug,
})
```

Will this be deduped/cached correctly? or do I need to do:
```tsx
const getPersonByIdCached = cache(getPersonById)

const preload = ({ params }: { params: { shortId: string; slug: string } }) => {
  getPersonByIdCached({
    shortId: params.shortId,
    slug: params.slug,
  })
}

// different component
getPersonByIdCached({
  shortId: params.shortId,
  slug: params.slug,
})
```

Any resources on this would be helpful also, I can't seem to find the answer for this. I remember the beta Next app docs having it the first way, but with the stable docs I only see the second way :(

**Comment by m1yon** [comment]
After some testing, you definitely have to do it the second way. Not sure why the beta docs used the first way as an example, the new stable docs are accurate though.

**Comment by zzzzzzxXxzzzzzz** [comment]
> In order for caching/deduping to work, do you have to call the same function returned from cache each time? Or is sending cache the same function good enough?
> 
> Example:
> ```tsx
> const preload = ({ params }: { params: { shortId: string; slug: string } }) => {
>   cache(getPersonById)({
>     shortId: params.shortId,
>     slug: params.slug,
>   })
> }
> 
> // different component
> cache(getPersonById)({
>   shortId: params.shortId,
>   slug: params.slug,
> })
> ```
> 
> Will this be deduped/cached correctly? or do I need to do:
> ```tsx
> const getPersonByIdCached = cache(getPersonById)
> 
> const preload = ({ params }: { params: { shortId: string; slug: string } }) => {
>   getPersonByIdCached({
>     shortId: params.shortId,
>     slug: params.slug,
>   })
> }
> 
> // different component
> getPersonByIdCached({
>   shortId: params.shortId,
>   slug: params.slug,
> })
> ```
> 
> Any resources on this would be helpful also, I can't seem to find the answer for this. I remember the beta Next app docs having it the first way, but with the stable docs I only see the second way :(
